src/dappsExplorer/search/DAppsExplorerScreenSearch.tsx

```diff
@@ -0,0 +1,327 @@
+
+  useEffect(() => {
+    dispatch(fetchDappsList())
+    ValoraAnalytics.track(DappExplorerEvents.dapp_screen_open)
+  }, [])
+
+  const allSectionResults: SectionData[] = React.useMemo(() => {
+    const allResultsParsed = parseResultsIntoAll(
+      dappListWithCategoryNames,
+      searchTerm,
+      favoriteDappsById
+    )
+    if (allResultsParsed.length === 0) {
+      setAllResultEmpty(true)
+    } else {
+      setAllResultEmpty(false)
+    }
+    return allResultsParsed
+  }, [dappList, searchTerm, favoriteDappsById])
+
+  const emptyListComponent = React.useMemo(() => {
+    if (allResultEmpty && favoriteResultsEmpty) return null
+    return <NoResultsSearch searchTerm={searchTerm} testID="DAppsExplorerScreenSearch" />
+  }, [allResultEmpty, favoriteResultsEmpty])
+
+  return (
+    <SafeAreaView
+      testID="DAppsExplorerScreenSearch"
+      style={styles.safeAreaContainer}
+      edges={['top']}
+    >
+      <DrawerTopBar
+        rightElement={
+          <TopBarIconButton
+            testID="DAppsExplorerScreenSearch/HelpIcon"
+            icon={<Help color={colors.onboardingGreen} />}
+            onPress={openSheet}
+          />
+        }
+        scrollPosition={scrollPosition}
+      />
+      {ConfirmOpenDappBottomSheet}
+      <>
+        {!loading && error && (
+          <View style={styles.centerContainer}>
+            <Text style={fontStyles.regular}>{t('dappsScreen.errorMessage')}</Text>
+          </View>
+        )}
+        {categories.length && (
+          <AnimatedSectionList
+            refreshControl={
+              <RefreshControl
+                tintColor={colors.greenBrand}
+                colors={[colors.greenBrand]}
+                style={styles.refreshControl}
+                refreshing={loading}
+                onRefresh={() => dispatch(fetchDappsList())}
+              />
+            }
+            // @ts-ignore TODO: resolve type error
+            ref={sectionListRef}
+            ListFooterComponent={
+              <>
+                {dappsMinimalDisclaimerEnabled ? (
+                  <Text style={styles.disclaimer}>{t('dappsDisclaimerAllDapps')}</Text>
+                ) : null}
+              </>
+            }
+            ListHeaderComponent={
+              <>
+                <DescriptionView
+                  title={t('dappsScreen.title')}
+                  message={t('dappsScreen.message')}
+                />
+                <SearchInput
+                  onChangeText={(value) => {
+                    setSearchTerm(value)
+                  }}
+                  value={searchTerm}
+                  multiline={false}
+                  placeholderTextColor={colors.gray4}
+                  underlineColorAndroid="transparent"
+                  placeholder={t('dappsScreen.searchPlaceHolder')}
+                  showClearButton={true}
+                  allowFontScaling={false}
+                />
+                {dappFavoritesEnabled && (
```

Comment:
Is it possible to be in that screen and `dappFavoritesEnabled` to not be `true`?

Reply:
🤔 I don't think so, but this does bring up a good point. I doubt we will be removing dapp favorites so it might be worth a clean-up pr to remove all the logic around if favorites is enabled and display it by default.

Reply:
Fixed in 6b9882a368accea810f9c533e9082c7fba72f920!